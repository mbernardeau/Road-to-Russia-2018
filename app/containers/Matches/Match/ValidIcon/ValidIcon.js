import React from 'react';
import PropTypes from 'prop-types';


import CheckIcon from 'material-ui/svg-icons/navigation/check';
import ClearIcon from 'material-ui/svg-icons/content/clear';

const iconStyles = {
  position: 'absolute',
  bottom: 5,
  left: 5,
  opacity: 0.8,
};

const ValidIcon = ({ valid }) => (valid ?
  <CheckIcon style={iconStyles} color="green" /> :
  <ClearIcon style={iconStyles} color="red" />
);

ValidIcon.propTypes = {
  valid: PropTypes.bool.isRequired,
};

export default ValidIcon;
