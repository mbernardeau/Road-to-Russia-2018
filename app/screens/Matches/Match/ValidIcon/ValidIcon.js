import React from 'react'
import PropTypes from 'prop-types'

import Tooltip from 'material-ui/Tooltip'

import CheckIcon from '@material-ui/icons/Check'
import ClearIcon from '@material-ui/icons/Clear'

const iconStyles = {
  position: 'absolute',
  top: 5,
  left: 5,
  opacity: 0.8,
}

const getTooltipText = valid => (valid ? 'Paris enregistrÃ©' : 'Paris invalide')

const ValidIcon = ({ valid }) => (
  <Tooltip title={getTooltipText(valid)} placement="right">
    {valid ? (
      <CheckIcon style={iconStyles} color="action" />
    ) : (
      <ClearIcon style={iconStyles} color="error" />
    )}
  </Tooltip>
)

ValidIcon.propTypes = {
  valid: PropTypes.bool.isRequired,
}

export default ValidIcon
