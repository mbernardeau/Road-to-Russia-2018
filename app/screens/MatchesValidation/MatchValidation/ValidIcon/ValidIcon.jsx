import React from 'react'
import PropTypes from 'prop-types'
import { onlyUpdateForPropTypes } from 'recompose'

import Tooltip from 'material-ui/Tooltip'

import CheckIcon from '@material-ui/icons/Check'
import ClearIcon from '@material-ui/icons/Clear'

import './ValidIcon.scss'

const ValidIcon = ({ valid }) => (
  <Tooltip title={valid ? 'Paris enregistrÃ©' : 'Paris invalide'} placement="right">
    <div className="tooltip-svg-container">
      {valid ? <CheckIcon className="valid-icon" /> : <ClearIcon className="invalid-icon" />}
    </div>
  </Tooltip>
)

ValidIcon.propTypes = {
  valid: PropTypes.bool.isRequired,
}

export default onlyUpdateForPropTypes(ValidIcon)
