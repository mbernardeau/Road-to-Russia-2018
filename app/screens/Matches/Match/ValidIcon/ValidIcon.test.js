import React from 'react'
import renderer from 'react-test-renderer'

import ValidIcon from './ValidIcon'

jest.mock('material-ui/Tooltip', () => 'Tooltip')
jest.mock('material-ui-icons/Check', () => 'Check')
jest.mock('material-ui-icons/Clear', () => 'Clear')

describe('screens/Match/ValidIcon', () => {
  it('renders correctly when valid', () => {
    const tree = renderer.create(<ValidIcon valid />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when invalid', () => {
    const tree = renderer.create(<ValidIcon valid={false} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
