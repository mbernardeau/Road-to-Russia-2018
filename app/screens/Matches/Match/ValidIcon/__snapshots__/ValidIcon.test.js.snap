// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens/Match/ValidIcon renders correctly when invalid 1`] = `
<Tooltip
  placement="right"
  title="Paris invalide"
>
  <Clear
    color="error"
    style={
      Object {
        "left": 5,
        "opacity": 0.8,
        "position": "absolute",
        "top": 5,
      }
    }
  />
</Tooltip>
`;

exports[`screens/Match/ValidIcon renders correctly when valid 1`] = `
<Tooltip
  placement="right"
  title="Paris enregistrÃ©"
>
  <Check
    color="action"
    style={
      Object {
        "left": 5,
        "opacity": 0.8,
        "position": "absolute",
        "top": 5,
      }
    }
  />
</Tooltip>
`;
